From ce2651dc8258ecd455c232b08504a8cdee8f3617 Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Mon, 16 Mar 2015 19:12:01 +0800
Subject: [PATCH 36/45] MA-6198 Write a common proxy properity for OMX

When Wifi connection was set a http proxy write
the proxy info into rw.HTTP_PROXY properity for
OMX Player.
The properity is clear to null when disconnect.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 .../java/com/android/server/ConnectivityService.java   | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/services/core/java/com/android/server/ConnectivityService.java b/services/core/java/com/android/server/ConnectivityService.java
index 327fb8a..73df6f5 100644
--- a/services/core/java/com/android/server/ConnectivityService.java
+++ b/services/core/java/com/android/server/ConnectivityService.java
@@ -4597,6 +4597,20 @@ public class ConnectivityService extends IConnectivityManager.Stub
                     " to " + state);
         }
 
+        if (state == NetworkInfo.State.CONNECTED) {
+            if (networkAgent.linkProperties.getHttpProxy() != null) {
+                String proxyHost = networkAgent.linkProperties.getHttpProxy().getHost();
+                String proxyPort = Integer.toString(networkAgent.linkProperties.getHttpProxy().getPort());
+                if (proxyHost != "") {
+                    log("we will set proxy to " + proxyHost + ":" + proxyPort + " in sendrolon");
+                    SystemProperties.set("rw.HTTP_PROXY", proxyHost + ":" + proxyPort);
+                } else {
+                    loge("Empty proxy host occured on makeDefault. Common proxy properity issued");
+                }
+            } else {
+                SystemProperties.set("rw.HTTP_PROXY", "");
+            }
+        }
         if (state == NetworkInfo.State.CONNECTED && !networkAgent.created) {
             try {
                 // This should never fail.  Specifying an already in use NetID will cause failure.
@@ -4755,6 +4769,10 @@ public class ConnectivityService extends IConnectivityManager.Stub
 
     protected void notifyNetworkCallbacks(NetworkAgentInfo networkAgent, int notifyType) {
         if (DBG) log("notifyType " + notifyTypeToName(notifyType) + " for " + networkAgent.name());
+        if (notifyType ==  ConnectivityManager.CALLBACK_LOST) {
+            if (DBG) log("As we lost networkAgent so clear common Proxy");
+            SystemProperties.set("rw.HTTP_PROXY", "");
+        }
         for (int i = 0; i < networkAgent.networkRequests.size(); i++) {
             NetworkRequest nr = networkAgent.networkRequests.valueAt(i);
             NetworkRequestInfo nri = mNetworkRequests.get(nr);
-- 
1.8.0

